packages/core/admin/ee/server/services/review-workflows/entity-service-decorator.js

```diff
@@ -0,0 +1,30 @@
+'use strict';
+
+const { hasRWEnabled, getDefaultWorkflow } = require('../../utils/review-workflows');
+
+/**
+ * Decorates the entity service with RW business logic
+ * @param {object} service - entity service
+ */
+const decorator = (service) => ({
+  async create(uid, opts = {}) {
+    const model = strapi.getModel(uid);
+    const hasRW = hasRWEnabled(model);
+
+    if (!hasRW) {
+      return service.create.call(this, uid, opts);
+    }
+
+    const defaultWorkFlow = await getDefaultWorkflow({ strapi });
+
+    return service.create.call(this, uid, {
+      ...opts,
+      // Assign this entity to the default workflow stage
+      data: { ...opts.data, strapi_reviewWorkflows_stage: defaultWorkFlow.stages[0].id },
+    });
```

Comment:
I would check if `opts.data` already has a `stage` assigned. 

If a user wants to set a specific stage, this would completely override the value.
```js
const data = { ...opts.data }

if (!has('strapi_reviewWorkflows_stage', data)) {
  data.strapi_reviewWorkflows_stage = await getDefaultWorkflow({ strapi });
 }
```

Reply:
or maybe using isNil instead of has🤔 so we don't allow the value to be nullish

Reply:
You could do another func 👍 
```javascript
async function getDataWithStage({ strapi }, data) {
  if (!has(ENTITY_ATTRIBUTE_NAME, data)) {
    const defaultWorkflow = await getDefaultWorkflow({ strapi });
    return { ...data, [ENTITY_ATTRIBUTE_NAME]: defaultWorkflow.stages[0].id };
  }
  return data;
}
```

Reply:
Thanks 🙏🏻 [`0c1f8a3` (#16122)](https://github.com/strapi/strapi/pull/16122/commits/0c1f8a3115e4bc9e83ea39a6a1f822734c940bc6)